internal/service/ec2/vpc_ipam_preview_next_cidr_test.go

```diff
@@ -0,0 +1,113 @@
+package ec2_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func TestAccVPCIpamPreviewNextCidr_ipv4Basic(t *testing.T) {
+	resourceName := "aws_vpc_ipam_preview_next_cidr.test"
+	netmaskLength := "28"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, ec2.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVPCIpamPreviewNextCidrIpv4Basic(netmaskLength),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "cidr"),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrPair(resourceName, "ipam_pool_id", "aws_vpc_ipam_pool.test", "id"),
+					resource.TestCheckResourceAttr(resourceName, "netmask_length", netmaskLength),
+				),
+			},
```

Comment:
I suggest adding a second `TestStep` with a config that has a second `aws_vpc_ipam_pool_cidr` resource declared and make `aws_vpc_ipam_preview_next_cidr.test` depend on this second CIDR. You should then check that `aws_vpc_ipam_preview_next_cidr.test` isn't recreated (by checking that the resource IDs are the same after the first and second steps).
This will test that the preview represents a snapshot of the next CIDR when the resource was created.

Reply:
Good suggestion. I'm wondering if a better representation of your idea is to create a vpc from the first cidr so its actually used and then validate that preview does not attempt to update itself because `--preview-next-cidr` command run via CLI would provide a new cidr value

note this will make acceptance tests for this resource take around 35m

Reply:
Resolved in a760780e0180d3f82fb55335dbadbedc27003212